sentry/src/db/campaign.rs

```diff
@@ -58,6 +62,99 @@ pub async fn fetch_campaign(
     Ok(row.as_ref().map(Campaign::from))
 }
 
+pub async fn list_campaigns(
+    pool: &DbPool,
+    skip: u64,
+    limit: u32,
+    creator: Option<Address>,
+    validator: Option<ValidatorId>,
+    is_leader: Option<bool>,
+    active_to_ge: &DateTime<Utc>,
+) -> Result<CampaignListResponse, PoolError> {
+    let client = pool.get().await?;
+
+    let (where_clauses, params) =
+        campaign_list_query_params(&creator, &validator, is_leader, active_to_ge);
+    let total_count_params = (where_clauses.clone(), params.clone());
+
+    // To understand why we use Order by, see Postgres Documentation: https://www.postgresql.org/docs/8.1/queries-limit.html
+    let statement = format!("SELECT campaigns.id, creator, budget, validators, title, pricing_bounds, event_submission, ad_units, targeting_rules, campaigns.created, active_from, active_to, channels.leader, channels.follower, channels.guardian, channels.token, channels.nonce FROM campaigns INNER JOIN channels ON campaigns.channel_id=channels.id WHERE {} ORDER BY campaigns.created ASC LIMIT {} OFFSET {}", where_clauses.join(" AND "), limit, skip);
+    let stmt = client.prepare(&statement).await?;
+    let rows = client.query(&stmt, params.as_slice()).await?;
+    let campaigns = rows.iter().map(Campaign::from).collect();
+
+    let total_count =
+        list_campaigns_total_count(pool, (&total_count_params.0, total_count_params.1)).await?;
+
+    // fast ceil for total_pages
+    let total_pages = if total_count == 0 {
+        1
+    } else {
+        1 + ((total_count - 1) / limit as u64)
+    };
+
+    let pagination = Pagination {
+        total_pages,
+        total: total_pages,
+        page: skip / limit as u64,
+    };
+
+    Ok(CampaignListResponse {
+        pagination,
+        campaigns,
+    })
+}
+
+fn campaign_list_query_params<'a>(
+    creator: &'a Option<Address>,
+    validator: &'a Option<ValidatorId>,
+    is_leader: Option<bool>,
+    active_to_ge: &'a DateTime<Utc>,
+) -> (Vec<String>, Vec<&'a (dyn ToSql + Sync)>) {
+    let mut where_clauses = vec!["active_to >= $1".to_string()];
+    let mut params: Vec<&(dyn ToSql + Sync)> = vec![active_to_ge];
+
+    if let Some(creator) = creator {
+        where_clauses.push(format!("creator = ${}", params.len() + 1));
+        params.push(creator);
+    }
+
+    // if clause for is_leader is true, the other clause is also always true
+    match (validator, is_leader) {
+        (Some(validator), Some(true)) => {
+            where_clauses.push(format!("channels.leader = ${}", params.len() + 1));
+            params.push(validator);
+        }
+        (Some(validator), _) => {
+            where_clauses.push(format!(
+                "(channels.leader = ${} OR channels.follower = ${})",
+                params.len() + 1,
+                params.len() + 1
```

Comment:
You are using the same prepared value (see `params.push(validator)` - only a single push to `params`) it should be enough to have 1 `params.len()`
```suggestion
                params.len() + 1,
```